File: typescript/sum-of-multiples/sum-of-multiples.ts
export function sum(divisors: number[], limit: number): number {
  return divisors
    .reduce((result: number[], divisor) => {
      for (let candidate = 1; candidate < limit; candidate++) {
        if (candidate % divisor === 0 && !result.includes(candidate)) {
          result.push(candidate);
        }
      }
      return result;
    }, [])
    .reduce((result, candidate) => result + candidate, 0);
}

File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

// SumMultiples finds the sum of all the unique multiples of particular numbers up to but not including that number.
func SumMultiples(limit int, divisors ...int) int {
	var known []int
	result := 0

	for _, divisor := range divisors {
	KnownLoop:
		for candidate := 1; candidate < limit; candidate++ {
			if candidate%divisor == 0 {
				for _, known := range known {
					if divisor == known {
						continue KnownLoop
					}
				}
				result += candidate
				known = append(known, candidate)
			}
		}
	}

	return result
}
